src/LanguageServer/Protocol/ExternalAccess/Razor/GetSymbolicInfoHandler.cs

```diff
@@ -0,0 +1,312 @@
+        SyntaxNode node,
+        TypeSyntax typeSyntax,
+        SemanticModel semanticModel,
+        PooledHashSet<AttributeSymbolicInfo> attributes,
+        ImmutableArray<ISymbol> writtenInsideBlockSymbols,
+        IEnumerable<IdentifierAndSymbol> identifiersInExpressions,
+        CancellationToken cancellationToken)
+    {
+        var declarationInfo = semanticModel.GetDeclaredSymbol(node, cancellationToken);
+        var typeSymbol = semanticModel.GetTypeInfo(typeSyntax, cancellationToken).Type;
+
+        if (declarationInfo is null || typeSymbol is null)
+        {
+            return;
+        }
+
+        var isWrittenTo = writtenInsideBlockSymbols.Any(symbol => SymbolEqualityComparer.Default.Equals(symbol, declarationInfo));
+
+        // Handle special case: attribute is string type or value type.
+        // Attributes of these types are not added to the 'WrittenInside' property of a data flow analysis when written to or mutated.
+
+        // Erring on the side of caution, assume they are written to if they are involved in some type of expression.
+
+        // The 'isWrittenTo' property is not critical to functionality in current usage; it's only used in ExtractToComponent
+        // to determine if a code attribute that has been promoted to a parameter in a component should include a comment warning.
+        if (typeSymbol.SpecialType == SpecialType.System_String || typeSymbol.IsValueType)
+        {
+            isWrittenTo = identifiersInExpressions.Any(symbol => SymbolEqualityComparer.Default.Equals(symbol.Symbol, declarationInfo));
+        }
+
+        attributes.Add(new AttributeSymbolicInfo
+        {
+            Name = declarationInfo.Name,
+            Type = GetFullTypeName(typeSyntax, semanticModel),
+            IsValueType = typeSymbol.IsValueType,
+            IsWrittenTo = isWrittenTo
+        });
+    }
+
+    private static string GetFullTypeName(TypeSyntax type, SemanticModel semanticModel)
+    {
+        var symbol = semanticModel.GetSymbolInfo(type).Symbol as ITypeSymbol;
+        if (symbol is not null)
+        {
+            return FormatType(symbol);
+        }
+
+        // Fallback to string if we can't get the symbol. Ideally this should never happen.
+        return type.ToString();
+    }
+
+    private static string FormatType(ITypeSymbol typeSymbol)
+    {
+        // Check if the symbol is a named type symbol (e.g., List<T>)
+        if (typeSymbol is INamedTypeSymbol namedTypeSymbol)
+        {
+            // Get the base name of the type, e.g., "List"
+            var typeName = namedTypeSymbol.Name;
+
+            // If there are type arguments, format them recursively
+            if (namedTypeSymbol.TypeArguments.Length > 0)
+            {
+                var typeArguments = string.Join(", ", namedTypeSymbol.TypeArguments.Select(FormatType));
+                return $"{typeName}<{typeArguments}>"; // Returning a formatted string seems hacky so might need to be revisited. 
```

Comment:
Best not to do this manually, but rather use a call to `ToDisplayString()` that specifies an appropriate set of `SymbolDisplayFormat` values to create the desired result.